docs/concepts/fundamentals/.nav.yml

```diff
@@ -1,15 +1,17 @@
 # `.nav.yml` is provided by https://lukasgeiter.github.io/mkdocs-awesome-nav
 nav:
   - Single-file components: single_file_components.md
-  - Components in Python: components_in_python.md
-  - Render API: render_api.md
+  - HTML / JS / CSS files: html_js_css_files.md
+  - HTML / JS / CSS variables: html_js_css_variables.md
+  - Secondary JS / CSS files: secondary_js_css_files.md
   - Component defaults: component_defaults.md
-  - Component context and scope: component_context_scope.md
+  - Render API: render_api.md
+  - Rendering components: rendering_components.md
```

Comment:
What was previously "Components in Python" was renamed to "Rendering components", and talks about all ways of rendering components.

What was previously "Defining HTML / JS / CSS files" was split to "HTML / JS / CSS files" and "Secondary JS / CSS files".

The page on "secondary JS / CSS" basically just talks about the `Media` class and how to define paths on it. I've split it off because it's quite info-dense topic.

Added page on variables / data methods like `get_template_data()`.